pkg/executor/adapter_slow_log.go

```diff
@@ -0,0 +1,183 @@
+// Copyright 2025 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor
+
+import (
+	"bytes"
+	"context"
+	"strings"
+	"time"
+
+	"github.com/pingcap/tidb/pkg/keyspace"
+	"github.com/pingcap/tidb/pkg/parser/ast"
+	"github.com/pingcap/tidb/pkg/sessionctx/variable"
+	"github.com/tikv/client-go/v2/util"
+)
+
+func init() {
+	variable.SlowLogRuleFieldAccessors[strings.ToLower(variable.SlowLogPlanDigest)] = variable.SlowLogFieldAccessor{
+		Parse: variable.ParseString,
+		Setter: func(_ context.Context, seVars *variable.SessionVars, items *variable.SlowQueryLogItems) {
+			_, planDigest := GetPlanDigest(seVars.StmtCtx)
+			items.PlanDigest = planDigest.String()
+		},
+		Match: func(_ *variable.SessionVars, items *variable.SlowQueryLogItems, threshold any) bool {
+			return variable.MatchEqual(threshold, strings.ToLower(items.PlanDigest))
+		},
+	}
+}
+
+// PrepareSlowLogItemsForRules builds a SlowQueryLogItems containing only the fields referenced by current session's SlowLogRules.
+// These pre-collected fields are later used for matching SQL execution details against the rules.
+func PrepareSlowLogItemsForRules(ctx context.Context, seVars *variable.SessionVars) *variable.SlowQueryLogItems {
```

Comment:
Will this function be called for each SQL statement? If yes, could it result in excessive small object allocations?

Reply:
Yes, if there is a slow log trigger rule.
Additionally,  only certain conditions within the rule will be retrieved; those that are not present will not be considered.